app/src/main/java/com/github/se/icebreakrr/model/notification/EngagementNotificationManager.kt

```diff
@@ -44,16 +48,29 @@ class EngagementNotificationManager(
     private val meetingRequestViewModel: MeetingRequestViewModel,
     private val appDataStore: AppDataStore,
     private val context: Context,
-    private val filterViewModel: FilterViewModel
+    private val filterViewModel: FilterViewModel,
+    private val permissionManager: IPermissionManager
 ) {
   private var notificationJob: Job? = null
   private val scope = CoroutineScope(Dispatchers.Main)
   private val lastNotificationTimes = mutableMapOf<String, Long>()
 
-  /** Start monitoring for nearby users with common tags */
+  /**
+   * Starts monitoring nearby users with common tags. Ensures POST_NOTIFICATIONS permission on
+   * Android TIRAMISU+ and launches monitoring in a coroutine.
+   */
   fun startMonitoring() {
     stopMonitoring() // Stop any existing monitoring
```

Comment:
Hmm, I'm not sure; I wasn't the one who wrote this code. However, calling this method before restarting monitoring seems like a good practice, even if it might ultimately be unnecessary.